website_compassion/models/res_user.py

```diff
@@ -31,3 +32,27 @@ def reset_password(self, login):
             if len(users) != 1:
                 raise
             return users.action_reset_password()
+
+
+    def _wp_url_gift_child(self, child):
+        self.ensure_one()
+        partner = self.partner_id.with_context(lang=self.lang)
+
+        if not child or not partner:
+            return "#"
+
+        data = {
+            "firstname": partner.preferred_name,
+            "pname": partner.name,
+            "email": partner.email,
+            "pstreet": partner.street,
+            "pcity": partner.city,
+            "pzip": partner.zip,
+            "pcountry": partner.country_id.name,
+            "sponsor_ref": partner.ref,
+            "child_ref": child.local_id,
+        }
+        filter_data = {key: val for key, val in data.items() if val}
+        params = urllib.parse.urlencode(filter_data)
+        url = f"{_('https://compassion.ch/de/geschenkformular/')}?{params}"
+        return url
```

Comment:
This is not needed. You can use the compute field already existing `partner.wordpress_form_data`

Reply:
I don't think I can. We want the link to be available for any child, with the suggested function it doesn't work if the sponsor have multiple children. See below

https://github.com/CompassionCH/compassion-switzerland/blob/4a47c9a99524d79072ee4d8a234e4c98ddb25adf/mass_mailing_switzerland/models/res_partner.py#L64

Reply:
I think it would be best to call, my new function with this condition to avoid code duplication

`child.local_id if len(child) == 1 else False
`

Reply:
You can put the child to use in the context variable `mailchimp_child` (I know, the name is not great, we could rename this variable if needed). It's read here : https://github.com/CompassionCH/compassion-switzerland/blob/4a47c9a99524d79072ee4d8a234e4c98ddb25adf/mass_mailing_switzerland/models/res_partner.py#L50